src/local-ssh/ipc/extension.ts

```diff
@@ -0,0 +1,259 @@
+import { WorkspaceInfo, WorkspaceInstancePhase } from '@gitpod/gitpod-protocol';
+import { ILogService } from '../../services/logService';
+import { ISessionService } from '../../services/sessionService';
+import { CallContext, ServerError, Status } from 'nice-grpc-common';
+import { IHostService } from '../../services/hostService';
+import { Server, createClient, createServer, createChannel } from 'nice-grpc';
+import { getDaemonVersion } from '../common';
+import { INotificationService } from '../../services/notificationService';
+import { showWsNotRunningDialog } from '../../remote';
+import { ITelemetryService, UserFlowTelemetry } from '../../services/telemetryService';
+import { ExperimentalSettings } from '../../experiments';
+import { SemVer } from 'semver';
+import { Configuration } from '../../configuration';
+
+const phaseMap: Record<WorkspaceInstanceStatus_Phase, WorkspaceInstancePhase | undefined> = {
+    [WorkspaceInstanceStatus_Phase.CREATING]: 'pending',
+    [WorkspaceInstanceStatus_Phase.IMAGEBUILD]: 'building',
+    [WorkspaceInstanceStatus_Phase.INITIALIZING]: 'initializing',
+    [WorkspaceInstanceStatus_Phase.INTERRUPTED]: 'running',
+    [WorkspaceInstanceStatus_Phase.PENDING]: 'stopping',
+    [WorkspaceInstanceStatus_Phase.PREPARING]: 'stopped',
+    [WorkspaceInstanceStatus_Phase.RUNNING]: 'running',
+    [WorkspaceInstanceStatus_Phase.STOPPED]: 'stopped',
+    [WorkspaceInstanceStatus_Phase.STOPPING]: 'stopping',
+    [WorkspaceInstanceStatus_Phase.UNSPECIFIED]: undefined,
+};
+
+export class ExtensionServiceImpl implements ExtensionServiceImplementation {
+    private notificationGapSet = new Set<string>();
+
+    constructor(private logService: ILogService, private sessionService: ISessionService, private hostService: IHostService, private notificationService: INotificationService, private experiments: ExperimentalSettings, private telemetryService: ITelemetryService) { }
+
+    private canShowNotification(id: string) {
+        if (this.notificationGapSet.has(id)) {
+            return false;
+        }
+        this.notificationGapSet.add(id);
+        setTimeout(() => {
+            this.notificationGapSet.delete(id);
+        }, 10000); // clean gap after 10s
+        return true;
+    }
+
+    async ping(_request: PingRequest, _context: CallContext): Promise<{}> {
+        return {};
+    }
+
+    async getWorkspaceAuthInfo(request: GetWorkspaceAuthInfoRequest, _context: CallContext): Promise<{ workspaceId?: string | undefined; workspaceHost?: string | undefined; ownerToken?: string | undefined }> {
+        try {
+            const accessToken = this.sessionService.getGitpodToken();
+            if (!accessToken) {
+                throw new ServerError(Status.INTERNAL, 'no access token found');
+            }
+            const workspaceId = request.workspaceId;
+
+            const gitpodHost = this.hostService.gitpodHost;
+            const usePublicApi = await this.experiments.getUsePublicAPI(gitpodHost);
```

Comment:
we agreed to use use public api for new features going forward but not sure which one has more priority to deploy to all users first local ssh or public api 🤔 
@akosyakov should we use just public api for this?

Reply:
We deploy it only for Gitpodders. Public API should be enabled for them already?

Reply:
yep, so lets remove the server api logic here @mustard-mh

Reply:
> yep, so lets remove the server api logic here @mustard-mh

In this case, for users, it'll only works with public api (no self-hosted support anymore?)

Reply:
for me it works both ways. I would like to try it with Gitpodders and learn more, we can figure out details later along the road.